File: php/features.php
<?php
declare(strict_types=1);

// ThurgauPopulationData SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ThurgauPopulationDataFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ThurgauPopulationDataBaseFeature();
            case "test":
                return new ThurgauPopulationDataTestFeature();
            default:
                return new ThurgauPopulationDataBaseFeature();
        }
    }
}

File: py/thurgaupopulationdata_sdk/features.py
# ThurgauPopulationData SDK feature factory

from thurgaupopulationdata_sdk.feature.base_feature import ThurgauPopulationDataBaseFeature
from thurgaupopulationdata_sdk.feature.test_feature import ThurgauPopulationDataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ThurgauPopulationDataBaseFeature(),
        "test": lambda: ThurgauPopulationDataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
